### PR TITLE
Fix Issue#55

### PR DESCRIPTION
**Pull request by skyguy94**
This PR fixes issue #55 by changing any year that occurs before 1980 to 1980 so that the archive gets created. It also outputs a warning about the date change during compression.

### CODE REVIEW

**Comment by KirkMunro** [comment]
Thank you for submitting this PR.

Before providing my review, I would like to hear from some others about the proposed changes.

@SteveL-MSFT: The APIs used to create an archive document the DateTime limits on the LastWriteTime attribute [here](https://msdn.microsoft.com/en-us/library/system.io.compression.ziparchiveentry.lastwritetime(v=vs.110).aspx). In the event when older documents are added to an archive, do we simply want to move forward their timestamp to within the allowed range, with a warning as is being done in this PR? I think it's a good compromise given the limits of the API being used for compression, although I would suggest the warning text should change to more clearly identify why it is warning the user, and I think the date should be set to the earliest possible date rather than just changing the year, but I'll handle those in a review after I hear your thoughts on this approach to this issue.

**Comment by SteveL-MSFT** [comment]
Given the limits of the api, I think a warning is sufficient.  Setting it to the earliest allowed date seems reasonable to me.

**Comment by skyguy94** [comment]
Thanks guys. I wasn't sure what to do exactly or what to log as I don't think this issue is all that straightforward to try and handle. The least complex thing I could think of was to bump up the date and warn.

I updated the PR to use `[DateTime]::Parse` and an ISO 8601 date with the default value in whatever the local time is for the machine performing the archiving. If you have any ideas about improving the log message let me know.

**Comment by KirkMunro** [comment]
Thanks Ritch. This is looking good. I have two main thoughts after reviewing the latest changes:

1. In order to accept this PR, we need Pester tests added to the [Pester test file](https://github.com/PowerShell/Microsoft.PowerShell.Archive/blob/master/Tests/Pester.Commands.Cmdlets.Archive.Tests.ps1) to make sure this works and is properly tested with Pester. Tests should specifically include files after 1980, where you validate that the LastWriteTime of the file before/after archiving (after extraction) is the same as it was in the beginning, and files before 1980, where you validate that the LastWriteTime after archiving is set to 1980-01-01T00:00:00. Make sense?
1. The warning message should do more than just tell the end user when a file has a LastWriteTime earlier than 1980. It should tell them why that is a warning. I also think the path should be in single quotes since it may contain spaces. Maybe something more like this:

    ```PowerShell
    Write-Warning "'$currentFilePath' has LastWriteTime earlier than 1980. Any files with LastWriteTime values earlier than 1980 will be stored in the archive with a LastWriteTime of 1980-01-01T00:00:00."
    ```

Would you please make those additional changes and then update the PR?

**Comment by skyguy94** [comment]
Ok, that test makes sense. For the parse, I can change it but I felt that being explicit about the time being the local 00:00 time. Without being explicit I felt it was unclear about zulu vs local time while reading the line.

**Comment by KirkMunro** [comment]
I'm ok with the explicit parse. You're right, it's less ambiguous.

**Comment by KirkMunro** [comment]
Given the rounding issue, I like your approach. The key point with this test was to verify that files with old dates had their dates properly updated post Expand-Archive, and that files with newer dates were unchanged, and using an inverted condition does that. Thanks for making those additional changes, Ritch! I'll do my final review now.

File: Tests/Pester.Commands.Cmdlets.Archive.Tests.ps1 [hunk 1]
```diff
@@ -1107,6 +1147,34 @@ Describe "Test suite for Microsoft.PowerShell.Archive module" -Tags "BVT" {
...
+        It "Validate that Compress-Archive/Expand-Archive work with dates earlier than 1980" {
+            $file1 = New-Item $TestDrive$($DS)SourceDir$($DS)EarlierThan1980.txt -Type File -Force   
+            $file1.LastWriteTime = [DateTime]::Parse('1974-10-03T04:30:00')
+            $file2 = "$TestDrive$($DS)SourceDir$($DS)Sample-1.txt"
+            $expandPath = "$TestDrive$($DS)EarlyYearDir"
+            $expectedFile1 = "$expandPath$($DS)EarlierThan1980.txt"
+            $expectedFile2 = "$expandPath$($DS)Sample-1.txt"
+            $archivePath = "$TestDrive$($DS)EarlyYear.zip"
+        
+            Write-Output "$expectedFile1"
```
Comment (KirkMunro) [inline Tests/Pester.Commands.Cmdlets.Archive.Tests.ps1]: Why Write-Output here? Was this put in for testing/troubleshooting and then left behind?

Comment (skyguy94) [reply Tests/Pester.Commands.Cmdlets.Archive.Tests.ps1]: Oops. yeah

**Comment by KirkMunro** [comment]
LGTM. Merging...

**Pull request merged**